src/components/Select/Select.tsx

```diff
@@ -0,0 +1,160 @@
+import * as React from 'react'
+import styled, { css } from 'styled-components'
+
+import { InjectedProps, withTheme } from '../../hocs/withTheme'
+
+export interface Options {
+  label: string
+  value: string
+}
+
+export interface Props {
+  className?: string
+  value: string
+  name: string
+  required?: boolean
+  disabled?: boolean
+  error?: boolean
+  width?: number | string
+  options: Options[]
+  blankOption?: boolean
+  placeholder?: string
+  onChange?: (name: string, value: string) => void
+}
+
+interface StyledProps extends InjectedProps {
+  width: string
+}
+
+interface SelectEvent {
+  currentTarget: {
+    value: string
+  }
+}
+
+class SelectComponent extends React.PureComponent<Props & InjectedProps> {
+  public render() {
+    const {
+      className = '',
+      value = '',
+      name,
+      required = false,
+      disabled = false,
+      error = false,
+      width = 260,
+      theme,
+      options,
+      blankOption = false,
+      placeholder = '',
+    } = this.props
+    const widthStyle = typeof width === 'number' ? `${width}px` : width
+    const classNames = `${className} ${error ? 'error' : ''}`
+
+    return (
+      <Wrapper width={widthStyle} theme={theme}>
+        <Base
+          className={classNames}
+          value={value}
+          name={name}
+          required={required}
+          disabled={disabled}
+          onChange={this.handleChange}
+          theme={theme}
+        >
+          {blankOption && <option value=""> {placeholder}</option>}
+          {options.map(option => (
+            <option key={option.value} value={option.value}>
+              {option.label}
+            </option>
+          ))}
+        </Base>
+      </Wrapper>
+    )
+  }
+
+  private handleChange = (e: SelectEvent) => {
+    const { name, onChange } = this.props
+    const value = e.currentTarget.value
+    if (onChange) onChange(name, value)
+  }
+}
+
+export const Select = withTheme(SelectComponent)
+
+const Wrapper = styled.div`
+  ${({ width, theme }: StyledProps) => {
+    const { palette } = theme
+    return css`
+      position: relative;
+      width: ${width};
+
+      &::before,
+      &::after {
+        position: absolute;
+        right: 12px;
+        width: 0;
+        height: 0;
+        padding: 0;
+        content: '';
+        border-left: 5px solid transparent;
+        border-right: 5px solid transparent;
+        pointer-events: none;
+      }
+
+      &::before {
+        border-top: 5px solid ${palette.TEXT_BLACK};
+        top: 52%;
+      }
+
+      &::after {
+        bottom: 52%;
+        border-bottom: 5px solid ${palette.TEXT_BLACK};
+      }
```

Comment:
[IMO] Implement icons with Icon Component for visual unification. There is an icon named 'fa-sort'.